pkg/operators/common/kubeinventorycache.go

```diff
@@ -110,10 +139,206 @@ func (cache *K8sInventoryCache) Stop() {
 	cache.useCount--
 }
 
-func (cache *K8sInventoryCache) GetPods() ([]*v1.Pod, error) {
-	return cache.pods.List(labels.Everything())
+// removeOldPods expects that the mutex is locked
+func (cache *inventoryCache) removeOldPods() {
+	now := time.Now()
+
+	toBeDeleted := make([]string, 0)
+	for key, oldPod := range cache.oldPods {
+		if now.Sub(oldPod.deletionTimestamp) > oldObjectTTL {
+			toBeDeleted = append(toBeDeleted, key)
+		}
+	}
+	for _, key := range toBeDeleted {
+		delete(cache.oldPods, key)
+	}
```

Comment:
No need to keep a `toBeDeleted` - you can call the `delete()` inside the first range operation.

Reply:
Good to know, thanks